src/App.tsx

```diff
@@ -1,35 +1,69 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './componets/Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+class App extends React.Component {
+  state = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+  nameInterval = 0;
 
-  // this code stops the timer
-  clearInterval(timerId);
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.stopClock);
+    document.addEventListener('click', this.startClock);
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    this.nameInterval = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-      <div className="clock">
-        <strong>{clockName}</strong>
-        {' time is '}
-        {date.toLocaleTimeString()}
+  componentDidUpdate(_: {}, prevState: { clockName: string }) {
+    if (prevState.clockName !== this.state.clockName) {
+      // eslint-disable-next-line no-console
+      console.log(`Renamed from ${prevState.clockName} to ${this.state.clockName}`);
+    }
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.stopClock);
+    document.removeEventListener('click', this.startClock);
+    clearInterval(this.nameInterval);
+  }
+
+  stopClock = () => {
+    this.setState({ hasClock: false });
+    clearInterval(this.nameInterval);
+  };
+
+  startClock = () => {
+    if (!this.state.hasClock) {
+      this.setState({ hasClock: true });
+
+      this.nameInterval = window.setInterval(() => {
+        this.setState({ clockName: getRandomName() });
+      }, 3300);
```

Comment:
don't repeat yourself, create a method for it

Reply:
fixed